### ISSUE TITLE
Feature Request: Refactor - Extract Method

### ISSUE THREAD

**Issue by scorpionlion** [PowerShell/vscode-powershell#1829]
**Summary of the new feature**
'As a developer I want a script segment to be extracted out as a function so that function can be used in other scripts files as well.'

**Proposed technical implementation details (optional)**
This is common functionality provided in VS code 
https://code.visualstudio.com/docs/editor/refactoring
https://refactoring.com/catalog/extractFunction.html

**Comment by mirind4** [comment]
I would love to have this!

**Comment by ravager-dk** [comment]
I second this.

**Comment by GlenAppleton** [comment]
Yes, please.  I've had this refactoring option for many other languages in different IDEs, and it's very useful.

**Comment by dpo007** [comment]
Yes, please! :)

**Comment by gRexGV** [comment]
IMO: This feature definitively needs attention. Or, according to documentation, may there is a plugin in the store which can provide the "extract-Method refactoring" for powershell-Files? I didn't find any.

**Comment by ErikApption** [comment]
the gods in chatgpt do not understand my requests either through copilot. Can't we get a simple extract method?
